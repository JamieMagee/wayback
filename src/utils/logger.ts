import { error as _error, warning as _warn } from '@actions/core';
import chalk from 'chalk';
// eslint-disable-next-line import/default
import stripAnsi from 'strip-ansi';

function write(prefix: string, ...args: unknown[]): void {
  console.log([`[${prefix}]`, ...args].join(' '));
}

function dir(obj: unknown): void {
  console.dir(obj);
}

function debug(msg: unknown, ...args: unknown[]): void {
  write(chalk.blue('DEBUG'), msg, ...args);
}

function info(msg: unknown, ...args: unknown[]): void {
  write(chalk.green(' INFO'), msg, ...args);
}

function warn(msg: unknown, ...args: unknown[]): void {
  write(chalk.magenta(' WARN'), msg, ...args);
  _warn(stripAnsi([msg, ...args].join(' ')));
}

function error(msg: unknown, ...args: unknown[]): void {
  write(chalk.red('ERROR'), msg, ...args);
  _error(stripAnsi([msg, ...args].join(' ')));
}

const log = (m: unknown, ...args: unknown[]): void => debug(m, ...args);

log.dir = (m: unknown): void => dir(m);
log.info = (m: unknown, ...args: unknown[]): void => info(m, ...args);
log.warn = (m: unknown, ...args: unknown[]): void => warn(m, ...args);
log.error = (m: unknown, ...args: unknown[]): void => error(m, ...args);

export default log;
