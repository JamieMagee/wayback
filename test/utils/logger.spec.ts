import * as _core from '@actions/core';
import chalk from 'chalk';
import log from '../../src/utils/logger';
import { getName, mocked } from '../utils';

jest.unmock('../../src/utils/logger');

const core = mocked(_core);

describe(getName(__filename), () => {
  const logger = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    console.log = logger;
    console.dir = logger;
  });

  it('works', () => {
    log('test');
    log.dir({ name: 'test', code: 1 });
    log.info(chalk.gray('test'), 'it');
    log.warn(chalk.yellow('test'), 'it');
    log.error(chalk.red('test'));

    expect(logger.mock.calls).toMatchSnapshot();
    expect(core.warning).toHaveBeenCalledWith('test it');
    expect(core.error).toHaveBeenCalledWith('test');
  });
});
