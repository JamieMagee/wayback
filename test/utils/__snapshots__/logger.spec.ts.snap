// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Array [
  Array [
    "[[34mDEBUG[39m] test",
  ],
  Array [
    Object {
      "code": 1,
      "name": "test",
    },
  ],
  Array [
    "[[32m INFO[39m] [90mtest[39m it",
  ],
  Array [
    "[[35m WARN[39m] [33mtest[39m it",
  ],
  Array [
    "[[31mERROR[39m] [31mtest[39m",
  ],
]
`;
